frontends/infinite-corridor/src/api/learning-resources/favorites.test.tsx

```diff
@@ -0,0 +1,107 @@
+import React from "react"
+import { renderHook } from "@testing-library/react-hooks/dom"
+import { QueryClient, QueryClientProvider } from "@tanstack/react-query"
+import { faker } from "@faker-js/faker"
+import { setMockResponse, act } from "../../test-utils"
+import { invalidateResourceQueries } from "./util"
+import { urls } from "./urls"
+import { clone } from "lodash"
+import { useInfiniteSearch } from "./search"
+import { useFavorite, useUnfavorite } from "./favorites"
+import { makeCourse, makeSearchResponse } from "ol-search-ui/src/factories"
+
+jest.mock("./util", () => {
+  const actual = jest.requireActual("./util")
+  return {
+    ...actual,
+    invalidateResourceQueries: jest.fn(actual.invalidateResourceQueries)
+  }
+})
+
+const setup = () => {
+  const queryClient = new QueryClient({
+    defaultOptions: {
+      queries: {
+        retry: false
+      }
+    }
+  })
+
+  const wrapper = ({ children }) => (
+    <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
+  )
+
+  const spies = {
+    queryClient: {
+      invalidateQueries: jest.spyOn(queryClient, "invalidateQueries")
+    },
+    invalidateResourceQueries: jest.mocked(invalidateResourceQueries)
+  }
+
+  return { wrapper, spies, queryClient }
+}
+
+describe.each([
+  {
+    wasFavorite: false,
+    useHook:     useFavorite,
+    favUrl:      urls.resource.favorite
+  },
+  {
+    wasFavorite: true,
+    useHook:     useUnfavorite,
+    favUrl:      urls.resource.unfavorite
+  }
+])("$useHook.name", ({ useHook, favUrl, wasFavorite }) => {
+  it("Invalidates resource queries", async () => {
+    const { wrapper } = setup()
+    const resource = makeCourse()
+
+    const { result } = renderHook(useHook, { wrapper })
+
+    setMockResponse.post(favUrl(resource.object_type, resource.id), resource)
+    await act(async () => {
+      await result.current.mutateAsync(resource)
+    })
+
+    expect(invalidateResourceQueries).toHaveBeenCalledWith(
```

Comment:
These tests are not new, though changed slightly (`invalidateResourceQueries` is introduced in this PR).

`learning-resources/hooks.ts` and the associated test file were getting a bit big, so I broke them into three:
- `resources.ts` for `useResource` and a few other queries (e.g., popular content)
- `resourceListings.ts` for userlist / stafflist things
- `favorites.ts`
and associated test files.